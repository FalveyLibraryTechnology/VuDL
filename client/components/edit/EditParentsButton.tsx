import React from "react";
import { useEditorContext } from "../../context/EditorContext";

export interface ObjectStatusProps {
    pid: string;
}

export const EditParentsButton = ({ pid }: ObjectStatusProps): React.ReactElement => {
    const {
        action: { setParentsModalActivePid, toggleParentsModal },
    } = useEditorContext();

    const clickAction = () => {
        setParentsModalActivePid(pid);
        toggleParentsModal();
    };
    return <button onClick={clickAction}>Edit Parents</button>;
};

export default EditParentsButton;
