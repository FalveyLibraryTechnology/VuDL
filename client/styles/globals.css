:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
