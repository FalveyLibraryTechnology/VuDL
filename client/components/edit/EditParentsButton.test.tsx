import React from "react";
import { describe, beforeEach, expect, it, jest } from "@jest/globals";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import EditParentsButton from "./EditParentsButton";

const mockUseEditorContext = jest.fn();
jest.mock("../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));

describe("EditParentsButton", () => {
    let editorValues;
    let pid: string;
    beforeEach(() => {
        pid = "foo:123";
        editorValues = {
            action: {
                setParentsModalActivePid: jest.fn(),
                toggleParentsModal: jest.fn(),
            },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
    });

    it("renders", () => {
        const wrapper = shallow(<EditParentsButton pid={pid} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("sets up modal on click", () => {
        const component = mount(<EditParentsButton pid={pid} />);
        component.find("button").simulate("click");

        expect(editorValues.action.setParentsModalActivePid).toHaveBeenCalledWith(pid);
        expect(editorValues.action.toggleParentsModal).toHaveBeenCalled();
        component.unmount();
    });
});
