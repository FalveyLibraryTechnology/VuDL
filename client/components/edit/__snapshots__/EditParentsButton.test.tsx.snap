// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditParentsButton renders 1`] = `
<button
  onClick={[Function]}
>
  Edit Parents
</button>
`;
